src/main/kotlin/telegram/Menu.kt

```diff
@@ -195,124 +284,375 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
+                        button(menuCommand named "Последняя", msgId, totalAvailablePages - 1)
                     }
                 }
+            } else {
+                button(blankCommand named "🤷 Пусто")
             }
-            button(adminBackCommand, messageId)
+            bottomContent(msgId)
         }
     )
 }
 
-internal fun showGameStatusMenu(chatId: Long, messageId: Long, bot: Bot) {
-    bot.editMessageReplyMarkup(
-        ChatId.fromId(chatId),
+internal fun <K: Any, T: Any> showPaginatedMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    title: String,
+    list: Collection<K, T>,
+    actionForEach: KeyboardContext.(T) -> Unit,
+    bottomContent: KeyboardContext.(Long) -> Unit,
+    menuCommand: Command,
+    desiredPageIndex: Int,
+    itemsPerPage: Int = defaultItemsPerPage
+) {
+    showPaginatedMenu(
+        chatId, messageId, bot, title,
+        list.find(),
+        actionForEach, bottomContent, menuCommand, desiredPageIndex, itemsPerPage
+    )
+}
+
+internal fun <K: Any, T: Any> showPaginatedAdminSubmenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    title: String,
+    list: Collection<K, T>,
+    actionForEach: KeyboardContext.(T) -> Unit,
+    menuCommand: Command,
+    pageIndex: Int,
+    itemsPerPage: Int = defaultItemsPerPage
+) {
+    showPaginatedMenu(
+        chatId,
         messageId,
-        replyMarkup = inlineKeyboard {
-            button(blankCommand named "Активные игры")
-            games.find().forEach {
-                button(blankCommand named it.name())
-                button(terminateGameCommand, it.id, messageId)
-            }
+        bot,
+        title,
+        list,
+        actionForEach,
+        {
             button(adminBackCommand, messageId)
-        }
+        },
+        menuCommand,
+        pageIndex,
+        itemsPerPage
     )
 }
 
-internal fun showHostAdminSettingsMenu(chatId: Long, messageId: Long, bot: Bot) {
-    bot.editMessageReplyMarkup(
-        ChatId.fromId(chatId),
+internal fun showHostAdminSettingsMenu(
+    chatId: Long,
+    messageId: Long,
+    bot: Bot,
+    pageIndex: Int = 0
+) {
+    showPaginatedAdminSubmenu(
+        chatId,
         messageId,
-        replyMarkup = inlineKeyboard {
-            button(blankCommand named "Ведущие")
-            hostSettings.find().forEach {
-                button(chooseHostAdminCommand named (it.host?.fullName()?: ""), messageId, it.hostId)
-            }
-            button(adminBackCommand, messageId)
-        }
+        bot,
+        "Ведущие",
+        hostSettings,
+        {
+            button(chooseHostAdminCommand named (it.host?.fullName()?: ""), -1L, it.hostId)
+        },
+        hostAdminSettingsCommand,
+        pageIndex
     )
 }
 
-internal fun showChosenSettingsMenu(chatId: Long, messageId: Long, bot: Bot, chosenId: Long) {
-    hostSettings.get(chosenId)?.let { settings ->
+fun sendMessage(
```

Comment:
Does this look like a menu? This function should not be in this file

What's more important I see no reason for moving bot form a function receiver to param. it just adds more lines to a git diff with no benefit.
Only reason `sendClosable` does not have a bot as a receiver is that it has context as a receiver, which is more beneficial.